### PR TITLE
[MRG] Packaging issues

### PR DESCRIPTION
**Pull request by mstimberg**
This PR fixes two issues:
* we previously shipped the Cython-generated C++ version of the spike queue as part of the source package which had the advantage that a pip install did not need Cython (it still needed a C++ compiler of course). The C++ code generated with older versions of Cython is not compatible with Python 3.7 (see #1026). I feel that it's the easiest to ship the Cython code only, and require a recent version of Cython as an install dependency (we want it for code generation on Python 3.x anyway). All this applies to the source installation only.
* As a first workaround to our issues with `py-cpuinfo` (#990, #1020), I removed it as a dependency for Linux and OS X (where we are not using it anyway), and made sure that it is only imported under Windows.

### CODE REVIEW

File: brian2/codegen/cpp_prefs.py [hunk 1]
```diff
@@ -7,31 +7,40 @@
...
+        # Note that this overwrites the arch_flag, i.e. only the best option will
+        # be used
+        if 'sse' in res['flags']:
+            msvc_arch_flag = '/arch:SSE'
+        if 'sse2' in res['flags']:
+            msvc_arch_flag = '/arch:SSE2'
+        if 'avx' in res['flags']:
+            msvc_arch_flag = '/arch:AVX'
+        if 'avx2' in res['flags']:
+            msvc_arch_flag = '/arch:AVX2'
```
Comment (thesamovar) [inline brian2/codegen/cpp_prefs.py]: I wonder if we could entirely eliminate our dependency on py-cpuinfo by just always having ``/arch:SSE /arch:SSE2 /arch:AVX /arch:AVX2`` in the flags for windows? It'll presumably ignore the ones it doesn't know but might use the last of the flags that are applicable? I haven't tried that though.

File: setup.py [hunk 2]
```diff
@@ -154,23 +154,26 @@ def build_extension(self, ext):
...
                         'pyparsing',
                         'jinja2>=2.7',
-                        'py-cpuinfo>=0.1.6, !=4.0.0',
-                        'setuptools>=6.0'  # FIXME: setuptools>=6.0 is only needed for Windows
+                        'py-cpuinfo>=0.1.6,!=4.0.0;platform_system=="Windows"',
+                        'setuptools>=21'
                        ],
       setup_requires=['numpy>=1.10',
-                      'setuptools>=6.0'
+                      'setuptools>=21'
```
Comment (thesamovar) [inline setup.py]: What's this?

File: brian2/codegen/cpp_prefs.py (hunk 1 above)
Comment (mstimberg) [reply brian2/codegen/cpp_prefs.py]: I'd be more than happy to do that, but we'd indeed need to test it first. Maybe you can do it on your machine, which I guess has AVX2, and I do it in the VM, which is probably on the lower end of capabilities?

File: setup.py (hunk 2 above)
Comment (thesamovar) [reply setup.py]: OK fine. Weird version numbers - they went all Chrome/Firefox style?

**Comment by mstimberg** [comment]
As just commented in the cpuinfo issues, I added a mini script that we can call to get the cpu flags. Since this script is protected with a `__main__` block, it should work fine with the latest version of cpuinfo.

This is executed once per `brian2` import (on Windows only), I did not bother to write the results to disk or anything like that. I'm not sure it's worth it given the potential complications.

**Comment by thesamovar** [comment]
This looks great, but I'd like to try it on my machine before merging.

How long does the cpuinfo subprocess launch take? If it's just a matter of a few ms then indeed not worth cacheing it.

**Comment by mstimberg** [comment]
> How long does the cpuinfo subprocess launch take? If it's just a matter of a few ms then indeed not worth cacheing it.

I only tested it under Linux, the script takes about 1s to execute. However, this is not the overhead from the subprocess launch but almost completely the time that importing cpuinfo and getting the information via `get_cpu_info` takes. It could be worth caching it, but not more so than before we switched to the subprocess approach.

**Comment by thesamovar** [comment]
If it's 1s I think cache it. If we save in ``~/.brian`` it's pretty safe I think?

**Comment by mstimberg** [comment]
> If it's 1s I think cache it

I think it does quite different things on Linux and on Windows, so it's worth checking it on Windows first (we are not using it at all on Linux/OS X now).

If we do the caching on-demand (as opposed to during installation), it should be pretty straight-forward. But even then, we should probably be careful and store some identifier of the machine or something like that with it. Running things on a university network with different machines but a shared home directory might otherwise be problematic (even though it is a quite low chance that the machines differ in the flags we care about).

**Comment by thesamovar** [comment]
1.4s on my reasonably powerful windows machine, so I think it's worth it. All tests pass too.

**Comment by thesamovar** [comment]
Let me know if/when you'd like me to do a final check.

**Comment by thesamovar** [comment]
OK code looks good and tests pass on my machine so I'm happy with it.

**Comment by mstimberg** [comment]
Great, thanks. I'll wait for the tests on the servers to pass and will then merge.

**Pull request merged**